### PR TITLE
feat: add godwoken readonly node and enable mem-pool p2p sync

### PR DESCRIPTION
**Pull request by sopium**

### CODE REVIEW

**Comment by sopium** [comment]
Manually tested OK with https://github.com/nervosnetwork/godwoken/pull/721/commits/5de77bf7e0214de5598501a4de9a3aaa38360b53

**Review by keroro520 — APPROVED** [review]

**Comment by sopium** [comment]
Integration test running at: https://github.com/nervosnetwork/godwoken/pull/721#issuecomment-1153590988

We should probably build a new image including https://github.com/nervosnetwork/godwoken/pull/721 before merging this.

**Review by Flouse — APPROVED** [review]

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -106,6 +106,35 @@ services:
       check-ckb-started-successfully:
         condition: service_completed_successfully
 
+  godwoken-readonly:
+    image: ghcr.io/keroro520/godwoken-prebuilds:v1-godwoken-d3eb41a-202205310512
```
Comment (Flouse) [inline docker/docker-compose.yml:110]:
```suggestion
    image ghcr.io/flouse/godwoken-prebuilds:1.2-gw-pull-721-merge-202206140118
```

Comment (Flouse) [reply docker/docker-compose.yml:110]: This image was built by https://github.com/Flouse/godwoken-docker-prebuilds/commit/f03dd561747d7ba7a093a07f51a66a1302dc25f6.

> Bump GODWOKEN_REF to [refs/pull/721/merge](https://github.com/nervosnetwork/godwoken/pull/721)

**Pull request merged**